"No module named 'pkg_resources'" error when installing against Python 3.12

On a fairly new* Python 3.12 virtualenv, running the following from the root of the Wagtail codebase

```
pip install -e .[docs]
cd docs
make html
```

results in the error:

```
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v5.3.0

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/Users/matthew/.virtualenvs/wagtailpy312/lib/python3.12/site-packages/sphinx/config.py", line 350, in eval_config_file
    exec(code, namespace)
  File "/Users/matthew/Development/tbx/wagtail/devscript/wagtail/docs/conf.py", line 19, in <module>
    import sphinx_wagtail_theme
  File "/Users/matthew/.virtualenvs/wagtailpy312/lib/python3.12/site-packages/sphinx_wagtail_theme/__init__.py", line 4, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'

make: *** [html] Error 2
```

This is with Python 3.12.0, pip 23.2.1.

\* created about a week ago - haven't tried replicating this from scratch yet

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@ All notable changes to this project will be documented in this file.
 ### Fixed
 
 -   Update `favicon` usage to be the new Sphinx variable `favicon_url` & add default favicon (Sahil Jangra)
+-   Avoid using pkgresources module for generating version, replace with use importlib.metadata (Salvo Polizzi)
 
 ### Security
 

--- a/CONTRIBUTORS.md
+++ b/CONTRIBUTORS.md
@@ -13,6 +13,7 @@
 -   Andy Chosak
 -   Temidayo Azeez
 -   Sahil Jangra
+-   Salvo Polizzi
 
 ## Sphinx Typo3 theme
 

--- a/sphinx_wagtail_theme/__init__.py
+++ b/sphinx_wagtail_theme/__init__.py
@@ -1,10 +1,10 @@
 """Sphinx Wagtail theme"""
 
 import os
-import pkg_resources
+from importlib.metadata import version
 
 
-__version__ = pkg_resources.require("sphinx_wagtail_theme")[0].version
+__version__ = version("sphinx-wagtail-theme")
 __version_full__ = __version__
 
 

